src/main/java/com/google/api/codegen/viewmodel/FileHeaderView.java

```diff
@@ -40,6 +40,12 @@ public boolean hasVersion() {
     return version() != null && version().length() > 0;
   }
 
+  public abstract String generatorVersion();
+
+  public boolean hasGeneratorVersion() {
+    return generatorVersion() != null && generatorVersion().length() > 0;
```

Comment:
I don't think we should add the complexity of caching. We don't have a performance problem in the generator. I'd like to revert this caching.